codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -236,7 +236,8 @@ private void writePendingBytes() {
 
     /**
      * This will allocate bytes by stream weight and priority for the entire tree rooted at {@code parent}, but does
-     * not write any bytes.
+     * not write any bytes. Priority is generally distributed proportionally to siblings in the priority tree, however
+     * we ensure that at least 1 byte is assigned to each stream in order to make progress.
```

Comment:
@Scottmitch yes it is arbitrary. All I really wanted to say is that it likely doesn't make any sense to allocate 1, 5 or 10 bytes of flow control window to a stream (again, arbitrary numbers). So maybe it would make more sense to implement the algorithm in a way that when the available flow control window is below a certain threshold, the bytes are no longer assigned proportionally to the weights but one  completes the highest priority streams first e.g. instead of assigning 10 bytes to 100 streams each, assign 1000 bytes to the highest priority stream, or 200 bytes to the 5 highest priority streams. Something like this.